dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -148,18 +173,115 @@ public ResourceSupport findRel(HttpServletRequest request, @PathVariable String
 		return findRelInternal(request, apiCategory, model, uuid, rel, page, assembler, projection);
 	}
 
-	@RequestMapping(method = RequestMethod.GET, value = "/{id:[A-z0-9]+}/{rel}/{relid:[A-z0-9]+}")
+	@RequestMapping(method = RequestMethod.GET, value = "/{id:^(?!^\\d+$)(?!^[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12}$)[\\w+\\-]+$}/{rel}/{relid:[\\w+\\-]+}")
 	public ResourceSupport findRel(HttpServletRequest request, @PathVariable String apiCategory,
 			@PathVariable String model, @PathVariable String id, @PathVariable String rel, @PathVariable String relid,
 			Pageable page, PagedResourcesAssembler assembler, @RequestParam(required = false) String projection) {
 		return findRelEntryInternal(request, apiCategory, model, id, rel, relid, page, assembler, projection);		
 	}
+
+	
+	@RequestMapping(method = RequestMethod.POST)
+	public ResponseEntity<ResourceSupport> post(HttpServletRequest request, @PathVariable String apiCategory,
+			@PathVariable String model) throws HttpRequestMethodNotSupportedException {		
+		return postInternal(request, apiCategory, model);
+	}
+	
+	public <ID extends Serializable> ResponseEntity<ResourceSupport> postInternal(HttpServletRequest request, String apiCategory,
+			String model) throws HttpRequestMethodNotSupportedException {
+		checkModelPluralForm(apiCategory, model);
+		DSpaceRestRepository<DirectlyAddressableRestModel, ID> repository = utils.getResourceRepository(apiCategory, model);
+		DirectlyAddressableRestModel modelObject = null;
+		try {
+			modelObject = repository.createAndReturn();
+		} catch (ClassCastException e) {
+			log.error(e.getMessage(), e);
+			return ControllerUtils.toEmptyResponse(HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		if (modelObject == null) {
+			throw new HttpRequestMethodNotSupportedException(RequestMethod.POST.toString());
+		}
+		DSpaceResource result = repository.wrapResource(modelObject);
+		//TODO manage HTTPHeader
+		return ControllerUtils.toResponseEntity(HttpStatus.CREATED, null, result);		
+	}	
+
+	@RequestMapping(method = RequestMethod.POST, value = "/{id:\\d+}" , headers = "content-type=multipart/form-data")
+	public <ID extends Serializable> ResponseEntity<ResourceSupport> upload(HttpServletRequest request,
+			@PathVariable String apiCategory, @PathVariable String model, @PathVariable Integer id,
+			@RequestParam(required=false, value="extraField") String extraField,
+		    @RequestParam("file") MultipartFile uploadfile) throws HttpRequestMethodNotSupportedException {
+		return uploadInternal(request, apiCategory, model, id, extraField, uploadfile);
+	}
+	
+	@RequestMapping(method = RequestMethod.POST, value = "/{uuid:[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12}}", headers = "content-type=multipart/form-data")	
```

Comment:
Please add a comment explaining this very large regex. It is not self-explanatory.